### PR TITLE
Add epic issue template

### PR DESCRIPTION
**Pull request by gtsiolis**
## Description

This will add an Epic template according to the [Epics](https://www.notion.so/gitpod/Development-Process-2-0-6681854173ab4d2f92880f9f3d85cab5#321619f5a4bd4391be83c66feb2cdb49) (internal) the new [Development Process](https://www.notion.so/gitpod/Development-Process-2-0-6681854173ab4d2f92880f9f3d85cab5) (internal) using Issue Forms[[1](https://gh-community.github.io/issue-template-feedback/structured/)]. Alternatively, instead of the issue form this could use plain markdown and inline all comments to  make the template more compact.

See how the [bug report](https://github.com/gitpod-io/gitpod/issues/new?assignees=&labels=bug&template=bug_report.yml) renders. See also [`bug_report.yml`](https://github.com/gitpod-io/gitpod/blob/gt/add-epic-issue-template/.github/ISSUE_TEMPLATE/bug_report.yml) and [`epic.yml`](https://github.com/gitpod-io/gitpod/blob/gt/add-epic-issue-template/.github/ISSUE_TEMPLATE/epic.yml).

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **gtsiolis** after the PR has been reviewed.

Associated issue: *#321619*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Opened this in case it's helpful during opening a new epic. Feel free to close if this looks unessesary. 📕 

Cc @csweichel @svenefftinge because [Development Process](https://www.notion.so/gitpod/Development-Process-2-0-6681854173ab4d2f92880f9f3d85cab5) (internal)

Cc @jldec @loujaybee @metcalfc because Product Sync[[1](https://www.notion.so/gitpod/Product-Sync-Dec-21-2021-d0fd22cad9bf43bb8703f961cd2766ce#8f0b0a9a98b140fd97dff018a70ed261)] (internal)

**Comment by lucasvaltl** [comment]
Reopening this after a discussion in our product sync. We want to use this as a baseline to iterate on an epic template that aims to help standardise epics and help epic creators think about a minimal set criteria that underline the value and intent of their epic.

File: .github/ISSUE_TEMPLATE/epic.yml
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/epic.yml:64]: Given we're not actively using persona's yet in the company, I'd suggest removing this for now. 

```suggestion
```

File: .github/ISSUE_TEMPLATE/epic.yml
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/epic.yml:104]: I'd suggest we remove this, I think the original intent for PRFAQ process was misunderstood, I don't think this makes sense to do as part of our epic template. 

```suggestion
```

**Comment by gtsiolis** [comment]
Thanks for re-opening this, @lucasvaltl!

Adding a few more points in favor of merging this as I've noticed some counter points for using a form here:
1. Using Issue Forms[[1](https://gh-community.github.io/issue-template-feedback/structured/)] can help structure our epics but also challenge what is an epic. From what I've seen having structure in the issue template always helps, for example see https://github.com/gitpod-io/gitpod/issues/3401, https://github.com/gitpod-io/gitpod/issues/3400, https://github.com/gitpod-io/gitpod/issues/3399, and more. These can give a hint on what feels unnecessary during epic creation and iterate by removing the required field if needed. 💭 
2. Required fields can be bypassed during epic creation by using a placeholder text like _N/A (Not applicable)_ or _TBD (To be discussed)_. 💡 
3. We can always change the form back to a plain markdown issue template and have the exact same rendering in the output but without the required fields at any point in the future if we feel like the form is adding too much friction.
4. Above all, try something. —Franklin D. Roosevelt

File: .github/ISSUE_TEMPLATE/epic.yml
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/epic.yml:48]: Measurement and hypothesis are quite closely linked, should we merge the two for simplicity? 

```suggestion
- type: textarea
  id: hypothesis-measurement
  attributes:
    label: Hypothesis & Measurement
    description: What behaviour change are we expecting to see in our user base? And how will that behaviour change be measured? Ideally this metric is one of our key product metrics.
    placeholder: Important as it's how we track the outcomes (not just output) of the work and prove a change was worth it. Or it should be removed or iterated.
  validations:
    required: true
```

File: .github/ISSUE_TEMPLATE/epic.yml
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/epic.yml:72]: Suggest removing, and merging with measurement [[1](https://github.com/gitpod-io/gitpod/pull/7556/files#r838382804)]

```suggestion
```

**Comment by gtsiolis** [comment]
**fyi**: Also, friendly reminder that the moment we change anything in this issue template, we'll be deviating from the initial structure defined in the [Development Process 2.0 / Epics](https://www.notion.so/gitpod/Development-Process-2-0-6681854173ab4d2f92880f9f3d85cab5#321619f5a4bd4391be83c66feb2cdb49) (internal), moving away from a [single source of truth (SSOT)](https://en.wikipedia.org/wiki/Single_source_of_truth). Not advocating against any changes but bringing this up for visibility as this also affects how we [ship skateboards](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#f7f54cb1c69643e9ac42c9f2ddd9e46f). ❗

**Review by loujaybee — APPROVED** [review]
Thanks @lucasvaltl and @gtsiolis ! 

Left a few comments to simplify and consolidate some fields. ✅ Approving as we can always iterate in future. ♻️ That said, I'll still personally continue to write out these templates as tables using [tablesgenerator.com](https://tablesgenerator.com/markdown_tables) (examples: #7570, #8513). 

Worth stressing that a huge aspect to these templates is 🧠 education and alignment. Creating the template is only the first tiny part. e.g. the `measurement` field needs to ideally be part of a cultural shift towards empowering teams with outcomes, rather than features. For as long as we're used to handing features to teams, but expecting teams to measure against outcomes, passing outcome measurements in this way will likely never feel natural. 

> Good process serves you so you can serve customers. But if you’re not watchful, the process can become the thing. [...] The process becomes the proxy for the result you want. You stop looking at outcomes and just make sure you’re doing the process right. Gulp [...] The process is not the thing. ([source](https://www.aboutamazon.com/news/company-news/2016-letter-to-shareholders)) 

🙏 Happy to ship 🛹  and iterate!

**Comment by lucasvaltl** [comment]
@loujaybee I actually quite like the changes you proposed because the reduce clutter and simplify the template, which makes it less overwhelming. To move forward on them, it might be best to create a new PR for your changes so we can discuss them and merge them as an iteration on what we have now? As @gtsiolis said, we'd have to make sure we keep [Development Process 2.0 / Epics](https://www.notion.so/gitpod/Development-Process-2-0-6681854173ab4d2f92880f9f3d85cab5#321619f5a4bd4391be83c66feb2cdb49) (internal) in sync.

**Pull request merged**
